lib/puppet/resource/catalog.rb

```diff
@@ -415,45 +420,34 @@ def resources
   def self.from_data_hash(data)
     result = new(data['name'], Puppet::Node::Environment::NONE)
 
-    if tags = data['tags']
-      result.tag(*tags)
-    end
-
-    if version = data['version']
-      result.version = version
-    end
-
-    if code_id = data['code_id']
-      result.code_id = code_id
-    end
-
-    if catalog_uuid = data['catalog_uuid']
-      result.catalog_uuid = catalog_uuid
-    end
-
+    result.tag(*data['tags']) if data['tags'] 
+    result.version = data['version'] if data['version']
+    result.code_id = data['code_id'] if data['code_id']
+    result.catalog_uuid = data['catalog_uuid'] if data['catalog_uuid']
     result.catalog_format = data['catalog_format'] || 0
 
-    if environment = data['environment']
+    environment = data['environment']
+    if environment
       result.environment = environment
       result.environment_instance = Puppet::Node::Environment.remote(environment.to_sym)
     end
 
-    if resources = data['resources']
-      result.add_resource(*resources.collect do |res|
-        Puppet::Resource.from_data_hash(res)
-      end)
-    end
+    result.add_resource(*data['resources'].collect do |res|
+      Puppet::Resource.from_data_hash(res)
+    end) if data['resources']
```

Comment:
I'm not a fan of the `if` hanging off the end of a multiline block. Prefer to see the usual:

```ruby
resources = data['resources']
if resources
  result.add_result(*resources...)
```